frontend/public/components/markdown-view.tsx

```diff
@@ -172,6 +173,8 @@ const IFrameMarkdownView: React.FC<InnerSyncMarkdownProps> = ({
   const [frame, setFrame] = React.useState<HTMLIFrameElement>();
   const [frameHeight, setFrameHeight] = React.useState(0);
   const [loaded, setLoaded] = React.useState(false);
+  const [theme, , themeLoaded] = useUserSettings<string>(THEME_USER_SETTING_KEY);
+  const localTheme = localStorage.getItem(THEME_LOCAL_STORAGE_KEY);
```

Comment:
Can you create a hook for this? So that we can use something like this here?

If **really** needed you can return the localStorage theme when loaded is false, but I don't think this is required for the markdown view?!

I would call it `useTheme` **only** if you also integrate the media query check into this hook.

```suggestion
  const [theme, themeLoaded] = useThemePreference();
```